libs/live-nft-react/src/hooks/useSpamTxFiltering.ts

```diff
@@ -0,0 +1,130 @@
+}) {
+  return useMemo(
+    () =>
+      spamFilteringTxEnabled
+        ? groupedOperations?.sections
+            .flatMap(section => section.data)
+            .reduce((acc, operation) => {
+              if (operation.type === "NFT_IN") {
+                const account = accountsMap[operation.accountId];
+                const contract = operation.contract || "";
+                acc[contract] = {
+                  contract,
+                  currencyId: getAccountCurrency(account).id,
+                };
+              }
+              return acc;
+            }, {} as NftOperations) || ({} as NftOperations)
+        : ({} as NftOperations),
+    [spamFilteringTxEnabled, groupedOperations, accountsMap],
+  );
+}
+
+export function filterSection(
+  operations: Operation[],
+  metadataMap: Map<string | undefined, number>,
+  accountsMap: Dictionary<AccountLike>,
+  markNftAsSpam?: (collectionId: string, blockchain: BlockchainsType, spamScore: number) => void,
+  threshold?: number,
+) {
+  const filtered: Operation[] = [];
+  operations.forEach(item => {
+    if (item.type === "NFT_IN") {
+      const spamScore = metadataMap.get(item.contract?.toLowerCase());
+      const ca = `${item.accountId}|${item.contract}`;
+      const account = accountsMap[item.accountId];
+      const currencyId = getAccountCurrency(account).id;
+
+      if (!spamScore || (threshold && spamScore <= threshold)) {
+        filtered.push(item);
+      } else {
+        markNftAsSpam?.(ca, currencyId as BlockchainEVM, spamScore);
+      }
+    } else {
+      filtered.push(item);
+    }
+  });
+
+  const uniqueOperationsMap = new Map(filtered.map(op => [op.id, op]));
+  const uniqueOperations = Array.from(uniqueOperationsMap.values());
+
+  return uniqueOperations;
+}
+
+export function filterGroupedOperations(
+  metadatas: Array<NFTResource<NonNullable<NFTCollectionMetadataResponse["result"]>>>,
+  accountsMap: Dictionary<AccountLike>,
+  groupedOperations?: DailyOperations,
+  markNftAsSpam?: (collectionId: string, blockchain: BlockchainsType, spamScore: number) => void,
+  threshold?: number,
+) {
+  const metadataMap = new Map(
+    metadatas.map(meta => [meta.metadata?.contract.toLowerCase(), meta.metadata?.spamScore ?? 0]),
+  );
+
+  return {
+    ...groupedOperations,
+    sections: groupedOperations?.sections.map(section => ({
+      ...section,
+      data: filterSection(section.data, metadataMap, accountsMap, markNftAsSpam, threshold),
+    })),
+  };
+}
+
+export function useSpamTxFiltering(
+  spamFilteringTxEnabled: boolean,
+  accountsMap: Dictionary<AccountLike>,
+  groupedOperations?: DailyOperations,
+  markNftAsSpam?: (collectionId: string, blockchain: BlockchainsType, spamScore: number) => void,
+  threshold?: number,
+) {
+  const nftOperations = useNftOperations({
+    accountsMap,
+    groupedOperations,
+    spamFilteringTxEnabled,
+  });
+
+  const metadatas = useNftCollectionMetadataBatch(Object.values(nftOperations));
```

Comment:
`Object.values` will also create a copy on each render. So I think this should be:
```suggestion
  const metadatas = useNftCollectionMetadataBatch(useMemo(() => Object.values(nftOperations), nftOperations));
```